src/Lean/Meta/Tactic/Apply.lean

```diff
@@ -231,6 +231,21 @@ def _root_.Lean.MVarId.apply (mvarId : MVarId) (e : Expr) (cfg : ApplyConfig :=
 def _root_.Lean.MVarId.applyConst (mvar : MVarId) (c : Name) (cfg : ApplyConfig := {}) : MetaM (List MVarId) := do
   mvar.apply (← mkConstWithFreshMVarLevels c) cfg (term? := m!"'{.ofConstName c}'")
 
+/-- Close the given goal using `e`, instantiated with `n` metavariables. -/
+def _root_.Lean.MVarId.applyN (mvarId : MVarId) (e : Expr) (n : Nat) (cfg : ApplyConfig := {}) : MetaM (List MVarId) :=
```

Comment:
The only thing that `cfg` is used for here is to control whether `isDefEq` uses the `approxDefEq` combinator.

I think it would be cleaner and easier to understand if `applyN` took `(useApproxDefEq : Bool := true)` instead. We could make this easy to use by modifying `approxDefEq` to have the definition
```lean
@[inline] def approxDefEq (x : n α) (useApproxDefEq : Bool := true) : n α :=
  if useApproxDefEq then
    mapMetaM approxDefEqImp x
  else
    x
```
(That's just one of the first things that came to mind.)

Reply:
I changed `isDefEqApply` to take `approx : Bool`.